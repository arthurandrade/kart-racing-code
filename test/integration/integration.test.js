const { recoverLaps } = require('./../../src/helper');
const Race = require('./../../src/kart/Race');

const processRace = () => {
  const laps = recoverLaps('race_log.txt');
  const race = new Race({ laps });

  return race.getResult();
};

describe('testing the integration', () => {
  test('should return race result', () => {
    const result = processRace();
    expect(result).toMatchSnapshot();
  });
});
