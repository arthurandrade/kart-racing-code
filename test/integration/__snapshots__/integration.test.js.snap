// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the integration should return race result 1`] = `
Array [
  Object {
    "avgLapSpeed": "44.246",
    "bestLap": 3,
    "bestLapTime": "1:2.769",
    "finishPosition": 1,
    "isFesterLap": true,
    "pilotId": "038",
    "pilotName": "F.MASS",
    "timeAfterFirstPilot": "0:0.000",
    "totalLaps": 4,
    "totalTime": "4:11.578",
  },
  Object {
    "avgLapSpeed": "43.627",
    "bestLap": 4,
    "bestLapTime": "1:3.076",
    "finishPosition": 2,
    "isFesterLap": false,
    "pilotId": "002",
    "pilotName": "K.RAIKKONEN",
    "timeAfterFirstPilot": "0:3.575",
    "totalLaps": 4,
    "totalTime": "4:15.153",
  },
  Object {
    "avgLapSpeed": "43.468",
    "bestLap": 3,
    "bestLapTime": "1:3.716",
    "finishPosition": 3,
    "isFesterLap": false,
    "pilotId": "033",
    "pilotName": "R.BARRICHELLO",
    "timeAfterFirstPilot": "0:4.502",
    "totalLaps": 4,
    "totalTime": "4:16.080",
  },
  Object {
    "avgLapSpeed": "43.191",
    "bestLap": 4,
    "bestLapTime": "1:4.216",
    "finishPosition": 4,
    "isFesterLap": false,
    "pilotId": "023",
    "pilotName": "M.WEBBER",
    "timeAfterFirstPilot": "0:6.144",
    "totalLaps": 4,
    "totalTime": "4:17.722",
  },
  Object {
    "avgLapSpeed": "38.066",
    "bestLap": 2,
    "bestLapTime": "1:7.011",
    "finishPosition": 5,
    "isFesterLap": false,
    "pilotId": "015",
    "pilotName": "F.ALONSO",
    "timeAfterFirstPilot": "0:42.643",
    "totalLaps": 4,
    "totalTime": "4:54.221",
  },
  Object {
    "avgLapSpeed": "25.746",
    "bestLap": 3,
    "bestLapTime": "1:18.097",
    "finishPosition": 6,
    "isFesterLap": false,
    "pilotId": "011",
    "pilotName": "S.VETTEL",
    "timeAfterFirstPilot": "2:15.698",
    "totalLaps": 3,
    "totalTime": "6:27.276",
  },
]
`;
